core/src/main/java/com/linecorp/armeria/internal/common/util/ThreadLocalInsecureRandom.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright 2021 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.common.util;
+
+import java.security.SecureRandom;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * Insecure {@link SecureRandom} which relies on {@link ThreadLocalRandom} for random number
+ * generation.
+ */
+final class ThreadLocalInsecureRandom extends SecureRandom {
```

Comment:
This is possibly technical debt inherited from netty but an insecure random implementing secure random is highly suspicious!

Lately I've learned that security scanners may make usage of a library at a large org difficult with this sort of thing (though removing the word insecure even without changing algo generally fixes it...

Reply:
The generation of self-signed certificates happens quite often while running tests, so using a real `SecureRandom` can exhaust system's entropy and make the tests block, so we intentionally use insecure random generator here.

I just inlined this into `SelfSignedCertificate` to make it a little bit more obscure :wink:

Reply:
Let me also inline `BouncyCastleSelfSignedCertGenerator` which seems small enough.